ts/nni_manager/core/nniTensorboardManager.ts

```diff
@@ -0,0 +1,218 @@
+// Licensed under the MIT license.
+
+'use strict';
+
+import * as fs from 'fs';
+import * as cp from 'child_process';
+import * as path from 'path';
+import { ChildProcess } from 'child_process';
+
+import * as component from '../common/component';
+import { getLogger, Logger } from '../common/log';
+import { getTunerProc, isAlive, killPid, uniqueString, mkDirPSync, getFreePort } from '../common/utils';
+import { Manager } from '../common/manager';
+import { TensorboardParams, TensorboardTaskStatus, TensorboardTaskInfo, TensorboardManager } from '../common/tensorboardManager';
+
+class TensorboardTaskDetail implements TensorboardTaskInfo {
+    public id: string;
+    public status: TensorboardTaskStatus;
+    public trialJobIdList: string[];
+    public trialLogDirectoryList: string[];
+    public pid?: number;
+    public port?: string;
+
+    constructor(id: string, status: TensorboardTaskStatus, trialJobIdList: string[], trialLogDirectoryList: string[]) {
+        this.id = id;
+        this.status = status;
+        this.trialJobIdList = trialJobIdList;
+        this.trialLogDirectoryList = trialLogDirectoryList;
+    }
+}
+
+class NNITensorboardManager implements TensorboardManager {
+    private log: Logger;
+    private tensorboardTaskMap: Map<string, TensorboardTaskDetail>;
+    private tensorboardVersion: string | undefined;
+    private nniManager: Manager;
+
+    constructor() {
+        this.log = getLogger();
+        this.tensorboardTaskMap = new Map<string, TensorboardTaskDetail>();
+        this.setTensorboardVersion();
+        this.nniManager = component.get(Manager);
+    }
+
+    public async startTensorboardTask(tensorboardParams: TensorboardParams): Promise<TensorboardTaskDetail> {
+        const trialJobIds = tensorboardParams.trials;
+        const trialJobIdList: string[] = [];
+        const trialLogDirectoryList: string[] = [];
+        await Promise.all(trialJobIds.split(',').map(async (trialJobId) => {
+            const trialTensorboardDataPath = path.join(await this.nniManager.getTrialOutputLocalPath(trialJobId), 'tensorboard');
+            mkDirPSync(trialTensorboardDataPath);
+            trialJobIdList.push(trialJobId);
+            trialLogDirectoryList.push(trialTensorboardDataPath);
+        }));
+        this.log.info(`tensorboard: ${trialJobIdList} ${trialLogDirectoryList}`);
+        return await this.startTensorboardTaskProcess(trialJobIdList, trialLogDirectoryList);
+    }
+
+    private async startTensorboardTaskProcess(trialJobIdList: string[], trialLogDirectoryList: string[]): Promise<TensorboardTaskDetail> {
+        const host = 'localhost';
+        const port = await getFreePort(host, 6006, 65535);
+        const command = await this.getTensorboardStartCommand(trialJobIdList, trialLogDirectoryList, port);
+        this.log.info(`tensorboard start command: ${command}`);
+        const tensorboardProc: ChildProcess = getTunerProc(command, 'ignore', process.cwd(), process.env, true, true);
```

Comment:
Will we throw an error on exception, e.g., tensorboard is not installed?

Reply:
add error log, and tensorboard installed check is in `setTensorboardVersion()`